packages/discord.js/src/client/websocket/WebSocketShard.js

```diff
@@ -565,7 +601,53 @@ class WebSocketShard extends EventEmitter {
     this.helloTimeout = setTimeout(() => {
       this.debug('Did not receive HELLO in time. Destroying and connecting again.');
       this.destroy({ reset: true, closeCode: 4009 });
-    }, 20_000).unref();
+    }, 20_000);
+  }
+
+  /**
+   * Sets the RESUMED dispatch timeout.
+   * @param {number} [time] If set to -1, it will clear the resumed timeout
+   * @private
+   */
+  setResumedDispatchTimeout(time) {
+    if (time === -1) {
+      if (this.resumedDispatchTimeout) {
+        this.debug('Clearing the RESUMED dispatch timeout.');
+        clearTimeout(this.resumedDispatchTimeout);
+        this.resumedDispatchTimeout = null;
+      }
+      return;
+    }
+    this.debug('Setting a RESUMED dispatch timeout for 20s.');
+    this.resumedDispatchTimeout = setTimeout(() => {
+      if ((Date.now() - this.lastReplayedAt) < 20_000) {
+        this.debug('Received a message within the last 20s. Delaying RESUMED timeout.');
+        return this.setResumedDispatchTimeout();
```

Comment:
https://eslint.org/docs/latest/rules/consistent-return
```suggestion
        this.setResumedDispatchTimeout();
        return
```